api/graphql/reservations/reservation_serializers.py

```diff
@@ -157,6 +157,31 @@ def validate_reservee_type(self, value):
             )
         return value
 
+    def validate_reservation(self, reservation_unit, scheduler, begin, end):
+        if reservation_unit.allow_reservations_without_opening_hours:
```

Comment:
returning here will mean we don't check `reservation_begins` dates and not check the overlap.

So I would remove this and add a condition in the `is_reservation_unit_open` like:
```
is_reservation_unit_open = reservation_unit.allow_reservations_without_opening_hours or scheduler.is_reservation_unit_open(begin, end)
```

But this thing should perhaps have its own method where no other than opening hour related would be checked (refs the above comment)